pynetdicom/dimse_messages.py

```diff
@@ -302,6 +303,12 @@ def decode_msg(self, primitive):
                         _MESSAGE_TYPES[self.command_set.CommandField][1]
                     )
 
+                    if self.__class__ == C_STORE and _config.STORE_RECV_CHUNKED_DATASET_PATH:
+                        self._data_set_file = (
+                            _config.STORE_RECV_CHUNKED_DATASET_PATH / f"{self.command_set.AffectedSOPInstanceUID}.dcm"
+                        )
+                        self.data_set = open(self._data_set_file, 'a+b')
```

Comment:
I'm not sure this is the best way to do this. For example, what happens if the SCU loses connection midway through? Or if an SCU sends the same dataset twice? We may end up having to create a temporary directory specific to an association and then copy completed datasets over to the destination directory.

The file should also be written in the DICOM file format.

Reply:
Is a FileIO the best choice here? Would it be better to do:
```python
with open(self._data_set_file, 'a+b') as f:
    f.write(data)
```
within the dataset writing section?

Reply:
Yeah I think you're right, probably better not to try to keep the file object around